### PR TITLE
[API] Project Subprojects

### PR DESCRIPTION
**Pull request by laemtl**
**API 0.0.4 dev** 
Adds the list of project's subprojects to the response of a /projects/$ProjectName GET request.

--------------

2.1 Single project

GET /projects/$ProjectName

Returns a 200 OK response if the project exists, and 404 Not Found if it does not (the same is true of any portion of the API under /projects/$ProjectName.)

The body of the request to /projects/$ProjectName will be an entity of the form:

{
    "Meta" : {
        "Project" : "ProjectName"
    },
    "Visits" : ["V1", "V2", ... ],
    "Instruments" : ["InstrumentName", "InstrumentName2", ...],
    "Candidates" : ["123543", "523234", ...],
    "Subprojects": ["SubprojectName1", "SubProhjectName2"]
}

-------------------------------------
To do:

- [ ] CHANGELOG.md
- [ ] modules/api/docs/LorisRESTAPI_v0.0.4-dev.md 
- [ ] test suite

### CODE REVIEW

**Comment by ridz1208** [comment]
@laemtl why does this not have a description ?

**Review by maltheism — APPROVED** [review]

**Review by driusan — CHANGES_REQUESTED** [review]
seems to be missing the api documentation and schema definition updates.

**Comment by christinerogers** [comment]
@xlecours if the API docs Changes Requested are covered by another PR already, could you please: 
* Link that PR here
* Update your review / dismiss the other change requests so that it's clear what changes still need to be made on this PR?

**Comment by christinerogers** [comment]
Now blocked by #7747

File: modules/api/php/views/project_0_0_4_dev.class.inc
Comment (driusan) [inline modules/api/php/views/project_0_0_4_dev.class.inc:40]: I think this key should be removed in order to decouple this from #7747 and unblock it. The rest of the results in this return seem uncontroversial and are all things that already exist in the API. Since it's still `-dev`, it's easier to add something to it if the other moves forward than remove it.

Comment (xlecours) [reply modules/api/php/views/project_0_0_4_dev.class.inc:40]: The whole point of this PR is to add that key.

Comment (driusan) [reply modules/api/php/views/project_0_0_4_dev.class.inc:40]: In that case I don't see why it's a separate PR. The whole point of the key seems to be to make the other one accessible.

Comment (xlecours) [reply modules/api/php/views/project_0_0_4_dev.class.inc:40]: Because at first, the subporjects were at the root of the api (like sites and projects and candidates)

**Comment by xlecours** [comment]
this will be integrated in https://github.com/aces/Loris/pull/7747

**Pull request closed without merging**
